types/adobe__pdfservices-node-sdk/execution-context.d.ts

```diff
@@ -0,0 +1,38 @@
+import { Credentials } from './auth/credentials';
+import { ClientConfig } from './client-config';
+
+/**
+ *
+ * Represents the execution context of an Operation. An execution context typically consists of the desired
+ * authentication credentials and client configurations such as timeouts.
+ * <p>
+ * For each set of credentials, a ExecutionContext instance can be reused across operations.
+ *
+ * Sample Usage:
+ * <pre class="prettyprint">
+ * <code>
+ * const credentials = PDFServicesSdk.Credentials.serviceAccountCredentialsBuilder()
+ *            .fromFile("pdfservices-api-credentials.json")
+ *            .build();
+ * const executionContext = PDFServicesSdk.ExecutionContext.create(credentials),
+ *        createPDFOperation = PDFServicesSdk.CreatePDF.Operation.createNew(),
+ *        input = PDFServicesSdk.FileRef.createFromLocalFile('files/resources/createPDFInput.docx');
+ *
+ * createPDFOperation.setInput(input);
+ *
+ * createPDFOperation.execute(executionContext)
+ *     .then(result => result.saveAsFile('output/createPDFFromZip.pdf'))
+ *     .catch(err => console.log(err));
+ * </code>
+ * </pre>
+ * @hideconstructor
+ *
+ */
+export class ExecutionContext {
```

Comment:
the reason for the no-unnecessary-class rule is that `create` returns a dangerous type: `ExecutionContext` doesn't have any properties, so anything is assignable to it:

```ts
let e = PDFServicesSDK.ExecutionContext.create(credentials)
let c = CombineFiles.Operation.createNew()
e = c // no error!
c.exec({}) // no error!
c.exec({ e: 1, m: 1}) // no error!
```

This is probably inherent to the design of the library, so you might just want to disable the rule. Alternatively, you could add a fake private property to make ExecutionContext nominal, something like `private __fake: never`.